airflow/jobs.py

```diff
@@ -1761,7 +1762,9 @@ def get_task_instances_for_dag_run(dag_run):
                 # can be more finegrained (excluding success or skipped)
                 tasks = {}
                 for ti in dag_run.get_task_instances():
-                    tasks[ti.key] = ti
+                    # the specified dag can be a subset of the whole dag (sub_dag)
```

Comment:
Nit: Might change to something like the following to be more clear # Need to filter on dag.task_ids because the specified dag might be a subset of the whole DAG and might not encompass all the task instances in the dagrun

It would be kinda nice if we could rework this so this wasn't a special case (i.e. dag_run.get_task_instances() would only return the relevant task instances) but this would probably be too much work for now.